### ISSUE TITLE
Make `team_maxRiflegrenades` only affect rifle grenades

### ISSUE THREAD

**Issue by Aciz** [etlegacy/etlegacy#1775]
Currently limiting rifle grenades with `team_maxRiflegrenades` will also limit the availability of the rifle weapon itself, not just the grenades. I don't see there ever being a scenario where limiting the rifle itself is required or wanted, so this cvar should probably be changed to only include rifle grenade limit.

**Comment by Helldembez** [comment]
Does this mean the amount of people that can select the weapon rifle, the amount of people that can load the rifle with a grenade or the amount of grenades that someone has when selecting the weapon rifle?

3 different things, but very confusing when talking about it. :D

**Comment by Aciz** [comment]
It would limit the amount of people that spawn and are able to collect ammo for rifle grenades.

**Comment by rmarquis** [comment]
See 5d01dcd55172ee5973b8da210871ad811dc42b73 (#1273) for current implementation, which afaik is the etpro one.
The proposed change above makes sense to me.

Edit: actually, have you looked at `team_maxRiflegrenades` vs `team_riflegrenades`? If I'm not mistaken, this is already possible?

**Comment by Aciz** [comment]
As a note, once implemented, the `team_riflegrenades` toggle for rifle grenades can be removed as you can then achieve the same thing with this cvar.

**Comment by rmarquis** [comment]
Ah, it seems you actually want to merge `team_maxRiflegrenades` with `team_riflegrenades` here.

I am not sure if that's a good idea. The `team_max` weapon cvars are part of the global weapon restriction feature (which some public servers make use of), while `team_riflegrenades` is more targeted to comp config. There *might* be some case where restricting the number of rifles is desired, without limiting the ability of using the grenade launchers - just not in competition.

To me, these are really two different features, and as such I don't think they should be merged.

**Comment by Aciz** [comment]
> There might be some case where restricting the number of rifles is desired, without limiting the ability of using the grenade launchers - just not in competition.

But this is currently not even possible? `team_maxRiflegrenades` is used to limit the amount of people that can spawn with an engineer rifle, not the amount of people that can spawn with rifle grenades. If you set it to say 2, then you can only have have 2 people spawning with engineer rifle at all. If you'd want to be able to limit number of rifles without limiting rifle grenades, you'd need a new cvar for that.

The reason that `team_riflegrenades` (just to make sure you remember, this is a *toggle*, not an integral limit) exists at all is so that on 3on3 configs, you can disable rifle grenades while still allowing people to spawn with rifle, just not with rifle grenades. If `team_maxRiflegrenades` would only limit the amount of people spawning with rifle grenades, you could just set it to 0 to achieve the same result.

**Comment by rmarquis** [comment]
Yes, `team_riflegrenades` is a toggle that enables or disables grenade launchers entirely. It's not actually possible to have a limit on both the number of engineer rifle and the number of rifle grenade launchers (for ex. 3 eng with rifle launcher, but 1 only allowed to use the grenade launcher).

A feature for allowing both (making `team_riflegrenades` not a `1/0` toggle, but a actual number of allowed grenade launchers) makes sense to me - much more than merging both cvars. However, the implementation might be somewhat tricky.

**Comment by Aciz** [comment]
Yeah @BystryPL pointed this out over at discord too. So, make `team_riflegrenades` and integral limit instead of a toggle, and rename this cvar to `team_maxRifles` then? This will then allow to limit just the rifle too if that's desired.

Although at that point, I think scrapping `team_riflegrenades` completely would make more sense (name wise), make `team_maxRiflegrenades` an integral limit of rifle grenades, and add `team_maxRifles` to control just the rifles.

**Comment by rmarquis** [comment]
Yup, sound perfect to me!

**Comment by BystryPL** [comment]
Fully agree with Big Boss @rmarquis here. As I simultanously suggested on Discord, I think better solution would be to change `team_riflegrenades` CVAR behavior to allow it to set an integral limit of riflegrenades per team and, on the other hand, keep `team_maxRiflegrenades` CVAR as it is. So keep 2 separate CVARs, one for rifles limit and one for riflegrenades limit, with the exception that t`eam_riflegrenades` value can be only equal or lower to `team_Maxriflegrenades` value.

Doing what Aciz suggest originally (merging two CVARs in one) might affect competitive gameplay in 2 ways:

1/ it wouldn't be possible to limit rifles anymore in 3on3 and 6on6 matches - and we do want to limit it to 1 per team, as it's a pretty powerful weapon if somebody knows how to shoot it, even without the riflegrenades.
2/ when using a script consisting in a bind tied to one key to toggle classes between engineer SMG/engineer RIFLE, it could cause accidentally picking up RIFLE instead of SMG as engineer's weapon (as rifles wouldn't be limited anymore), which is not desirable either.
